### ISSUE TITLE
CI using Travis, etc?

### ISSUE THREAD

**Issue by oliverlockwood** [cyclestreets/android-mappacks#4]
@jezhiggins @mvl22 would it be useful to set up CI builds of this on Travis, publishing betas to the Play Store as we do for the android app itself?

If so, I'd be happy to take this on.  Let me know your thoughts.

**Comment by jezhiggins** [comment]
It would.  Part of the reason I hadn't previously got this going was that Mapsforge doesn't use standard OSM data.  It has its own file format, and generating that as part of the build took hours and was prone to running out of memory (almost regardless of how much was available).  

Mapsforg (and others) are now providing maps, so we can sidestep this whole palaver.

I'll update the build accordingly, and then we can get Travis going.

**Comment by oliverlockwood** [comment]
Good stuff. I'll wait for your changes to go in before getting started, then.

**Comment by mvl22** [comment]
> Mapsforge (and others) are now providing maps, so we can sidestep this whole palaver.

Can I check whether these days there is a better mechanism for distributing pre-computed map packs? It's always felt a little odd to me that these are in the form of a separate quasi-app, which feels rather clunky and unusual. I think we're doing that because Google is hosting that for free, and they are large size thus saving costs. But increasingly I feel we could absorb this in our current hosting, and therefore just have a simple downloading process within the app itself.

**Comment by jezhiggins** [comment]
Packaging the maps as separate apps in unusual, but it's not entirely without precedent. If we took the downloading into the app and did ourselves, there are a number of features we'd either lose or have to build ourselves

- automatic checking for updates
- resumable downloads
- user controllable download over mobile network or wifi only
- beta releases

**Comment by mvl22** [comment]
> there are a number of features we'd either lose or have to build ourselves

I agree, but I think internalising this within the main app is surely where we should try to be heading. Currently we have two packs - UK and Prague, but nowadays the routing covers other areas. It will be silly to have to publish lots of packs - it's not a very scalable model and I think it's rather confusing for end-users.

Just trying to enumerate that list a little better:

> automatic checking for updates

That presumably is some kind of HTTP or API call, so I guess not too hard.

> resumable downloads

I agree this would be a bit of work, but I wonder if there is a library that does this kind of thing? It can't be that uncommon.

> user controllable download over mobile network or wifi only

I would have thought that is just a setting that we then check.

> beta releases

That presumably isn't an issue as it would just be part of the main app, and we already have the Beta publishing in place.

It would be good if we could scope out whether this really would be all that difficult?

**Comment by oliverlockwood** [comment]
I should be able to get a few hours on this tonight; I'll try to get my head around the whole setup and determine what our options are going forward.
